Assets/Scripts/UnitController.cs

```diff
@@ -59,6 +63,29 @@ void FixedUpdate()
 
     void UnitMovement()
     {
+        if (IsOffScreenVertically(border1Sprite) && IsOffScreenVertically(border2Sprite) && IsOffScreenVertically(border3Sprite) && IsOffScreenVertically(border4Sprite))
```

Comment:
Why check if the all the borders of the unit are offscreen instead of just the main sprite? Seems unnecessary